AmplifyPlugins/API/AWSAPICategoryPlugin/Support/Utils/GraphQLResponseDecoder.swift

```diff
@@ -53,12 +53,17 @@ struct GraphQLResponseDecoder {
 
         case (.some(let data), .some(let errors)):
             do {
-                let jsonValue = JSONValue.object(data)
-                let responseData = try decode(graphQLData: jsonValue,
-                                              into: responseType,
-                                              at: decodePath)
                 let responseErrors = try decodeErrors(graphQLErrors: errors)
-                return GraphQLResponse<R>.failure(.partial(responseData, responseErrors))
+
+                do {
+                    let jsonValue = JSONValue.object(data)
+                    let responseData = try decode(graphQLData: jsonValue,
+                                                  into: responseType,
+                                                  at: decodePath)
+                    return GraphQLResponse<R>.failure(.partial(responseData, responseErrors))
+                } catch let decodingError as DecodingError {
+                    return GraphQLResponse<R>.failure(.error(responseErrors))
+                }
```

Comment:
I think it make the code more readable if we avoided nested do/catch statements... what do you think?

Reply:
+1. I'd also favor breaking out the handling into a separate method after it grows beyond about 3 lines.

Reply:
thanks, will do another pass on refactoring this

Reply:
updated code, pulled out the nested try-catch into sequential logic so no longer a nested try-catch block. Couldn't think of a quick way it out into separate methods and name them accurately and will leave that for another time (still have TODO in that file to do separate methods)